src/test/java/uk/nhs/hee/tis/trainee/forms/model/AbstractAuditedFormTest.java

```diff
@@ -66,6 +69,118 @@ void shouldReturnNullSubmittedWhenNoStatus() {
+    Person modifiedBy = new Person("name", "email", "role");
+    form.setLifecycleState(LifecycleState.SUBMITTED, detail, modifiedBy, 2);
+
+    assertEquals(LifecycleState.SUBMITTED, form.getLifecycleState(),
+        "Unexpected lifecycle state.");
+    Status.StatusInfo statusInfo = form.getStatus().current();
+    assertEquals(LifecycleState.SUBMITTED, statusInfo.state(),
+        "Unexpected lifecycle state in status.");
+    assertThat("Unexpected modified by in status.", statusInfo.modifiedBy(), is(modifiedBy));
+    assertThat("Unexpected detail in status.", statusInfo.detail(), is(detail));
+    assertThat("Unexpected revision in status.", statusInfo.revision(), is(2));
+    assertThat("Unexpected timestamp in status.",
+        statusInfo.timestamp().truncatedTo(ChronoUnit.SECONDS),
+        is(Instant.now().truncatedTo(ChronoUnit.SECONDS)));
+
+    Status.StatusInfo statusInfoHistory = form.getStatus().history().get(0);
+    assertEquals(statusInfoHistory, statusInfo, "Unexpected status info in history.");
+  }
+
+  @Test
+  void shouldSetLifecycleStateIfCurrentIsMissing() {
+    form.setRevision(1);
+    form.setStatus(new Status(null, List.of()));
+    form.setLifecycleState(LifecycleState.SUBMITTED);
+
+    assertEquals(LifecycleState.SUBMITTED, form.getLifecycleState(),
+        "Unexpected lifecycle state.");
+    Status.StatusInfo statusInfo = form.getStatus().current();
+    assertEquals(LifecycleState.SUBMITTED, statusInfo.state(),
+        "Unexpected lifecycle state in status.");
+    assertThat("Unexpected modified by in status.", statusInfo.modifiedBy(), nullValue());
+    assertThat("Unexpected detail in status.", statusInfo.detail(), nullValue());
+    assertThat("Unexpected revision in status.", statusInfo.revision(), is(1));
+    assertThat("Unexpected timestamp in status.",
+        statusInfo.timestamp().truncatedTo(ChronoUnit.SECONDS),
+        is(Instant.now().truncatedTo(ChronoUnit.SECONDS)));
+
+    Status.StatusInfo statusInfoHistory = form.getStatus().history().get(0);
+    assertEquals(statusInfoHistory, statusInfo, "Unexpected status info in history.");
+  }
+
+  @Test
+  void shouldOverwriteExistingLifecycleState() {
+    form.setLifecycleState(LifecycleState.DRAFT);
+    form.setLifecycleState(LifecycleState.APPROVED);
+
+    assertEquals(LifecycleState.APPROVED, form.getLifecycleState(),
+        "Unexpected lifecycle state.");
+  }
+
+  @Test
+  void shouldHandleNullLifecycleState() {
+    form.setLifecycleState(null);
+
+    assertNull(form.getLifecycleState(), "Expected lifecycle state to be null.");
+  }
+
+  @Test
+  void shouldAddLifecycleStateToHistory() {
+    form.setLifecycleState(LifecycleState.DRAFT);
+    form.setLifecycleState(LifecycleState.SUBMITTED);
+
+    assertThat("Unexpected status history items.", form.getStatus().history().stream()
+        .map(StatusInfo::state).toList()
+        .containsAll(List.of(DRAFT, SUBMITTED)));
```

Comment:
This should be changed to `is(List.of(DRAFT, SUBMITTED))` to verify that they are correctly ordered.

Reply:
The ordering is SUBMITTED, DRAFT since it pushes them in to the top of the list - is that correct? We seem to order them by date when they are retrieved regardless of how they are listed.

Reply:
We do, so it doesn't _really_ matter too much, but imo it makes more sense to add changes to the end of the list. So the latest is the highest index.